Code/BinkMovie/FFMpegPlayer.h

```diff
@@ -0,0 +1,72 @@
+/*
+**	Command & Conquer Renegade(tm)
+**	Copyright 2025 OpenW3D.
+**
+**	This program is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 3 of the License, or
+**	(at your option) any later version.
+**
+**	This program is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include "rect.h"
+#include "render2dsentence.h"
+#include "wwstring.h"
+#include <memory>
+#include <vector>
+
+class FontCharsClass;
+class SubTitleManagerClass;
+class FFmpegFile;
+struct AVFrame;
+struct SwsContext;
+
+class FFMpegMovieClass
```

Comment:
The renegade code heavily uses inheritance. Would it make sence to create an abstract `MovieClass`, which the null player, bink player and ffmpeg player inherit?
Then, we can build a x86 renegade exe that supports multiple decoders, which might be useful for debugging and adding support for other decoders.

Reply:
I've implemented most of these suggestions with the exception that I don't have a runtime selection of the decoder backend (though in theory that is now supportable).